mslib/msui/mscolab.py

```diff
@@ -276,21 +275,68 @@ def login_handler(self):
         else:
             self.mscolab.after_login(emailid, self.mscolab_server_url, r)
 
+    def new_user_login_handler(self, emailid, password):
+        for key, value in config_loader(dataset="MSC_login").items():
+            if key not in constants.MSC_LOGIN_CACHE or constants.MSC_LOGIN_CACHE[key] != value:
+                constants.MSC_LOGIN_CACHE[key] = value
+        auth = constants.MSC_LOGIN_CACHE.get(self.mscolab_server_url, (None, None))
+
+        data = {
+            "email": emailid,
+            "password": password
+        }
+        s = requests.Session()
+        s.auth = (auth[0], auth[1])
+        s.headers.update({'x-test': 'true'})
+        url = f'{self.mscolab_server_url}/token'
+        try:
+            r = s.post(url, data=data)
+        except requests.exceptions.ConnectionError as ex:
+            logging.error("unexpected error: %s %s %s", type(ex), url, ex)
+            self.set_status(
+                "Error",
+                "Failed to establish a new connection" f' to "{self.mscolab_server_url}". Try in a moment again.',
+            )
+            self.disconnect_handler()
+            return
+        data_to_save_in_config_file = {
+            "MSCOLAB_mailid": emailid,
+            "MSCOLAB_password": password
+        }
+        if config_loader(dataset="MSCOLAB_mailid") == "" and config_loader(dataset="MSCOLAB_password") == "":
+            modify_config_file(data_to_save_in_config_file)
```

Comment:
the test runs in an error now

```
2022-02-16 19:25:13 CRITICAL Fatal error: Traceback (most recent call last):
  File "/home/reimar/PycharmProjects/2022/sr1j/MSS/mslib/msui/mscolab.py", line 376, in new_user_handler
    self.new_user_login_handler(emailid, password)
  File "/home/reimar/PycharmProjects/2022/sr1j/MSS/mslib/msui/mscolab.py", line 307, in new_user_login_handler
    modify_config_file(data_to_save_in_config_file)
  File "/home/reimar/PycharmProjects/2022/sr1j/MSS/mslib/utils/config.py", line 409, in modify_config_file
    modified_data = merge_data(json_file_data_copy, data)
  File "/home/reimar/PycharmProjects/2022/sr1j/MSS/mslib/utils/config.py", line 553, in merge_data
    data, match = compare_data(options[key], json_file_data[key])
KeyError: 'MSCOLAB_mailid'
```

Reply:
the functional tests trying your changes gives for me the wanted result :)

But have also a look on the tests because, we try to protect by tests given functionality.

Reply:
Moin @ReimarBauer, yep I was looking into the tests.

Reply:
I'll take a look